cli/flox-activations/src/cli/fix_paths.rs

```diff
@@ -0,0 +1,216 @@
+    if !joined.ends_with(':') {
+        joined.push(':');
+    }
+    joined
+}
+
+#[cfg(test)]
+mod test {
+    use std::str::FromStr;
+
+    use super::*;
+
+    #[test]
+    fn appends_suffixes() {
+        let suffixes = ["suffix"];
+        let flox_env_dirs = [PathBuf::from_str("/flox_env").unwrap()];
+        let path_dirs = [
+            PathBuf::from_str("/path1").unwrap(),
+            PathBuf::from_str("/path2").unwrap(),
+        ];
+        let new_dirs = prepend_dirs_to_pathlike_var(&flox_env_dirs, &suffixes, &path_dirs);
+        assert_eq!(new_dirs[0], PathBuf::from_str("/flox_env/suffix").unwrap());
+    }
+
+    #[test]
+    fn flox_envs_added_in_order() {
+        let suffixes = ["suffix"];
+        let flox_env_dirs = [
+            PathBuf::from_str("/flox_env1").unwrap(),
+            PathBuf::from_str("/flox_env2").unwrap(),
+        ];
+        let path_dirs = [
+            PathBuf::from_str("/path1").unwrap(),
+            PathBuf::from_str("/path2").unwrap(),
+        ];
+        let new_dirs = prepend_dirs_to_pathlike_var(&flox_env_dirs, &suffixes, &path_dirs);
+        assert_eq!(new_dirs, vec![
+            PathBuf::from_str("/flox_env1/suffix").unwrap(),
+            PathBuf::from_str("/flox_env2/suffix").unwrap(),
+            PathBuf::from_str("/path1").unwrap(),
+            PathBuf::from_str("/path2").unwrap(),
+        ]);
+    }
+
+    #[test]
+    fn duplicate_paths_removed() {
+        let suffixes = ["suffix"];
+        let flox_env_dirs = [
+            PathBuf::from_str("/flox_env1").unwrap(),
+            PathBuf::from_str("/flox_env1").unwrap(),
+            PathBuf::from_str("/flox_env2").unwrap(),
+            PathBuf::from_str("/flox_env2").unwrap(),
+        ];
+        let path_dirs = [
+            PathBuf::from_str("/path1").unwrap(),
+            PathBuf::from_str("/path1").unwrap(),
+            PathBuf::from_str("/path2").unwrap(),
+            PathBuf::from_str("/path2").unwrap(),
+        ];
+        let new_dirs = prepend_dirs_to_pathlike_var(&flox_env_dirs, &suffixes, &path_dirs);
+        assert_eq!(new_dirs, vec![
+            PathBuf::from_str("/flox_env1/suffix").unwrap(),
+            PathBuf::from_str("/flox_env2/suffix").unwrap(),
+            PathBuf::from_str("/path1").unwrap(),
+            PathBuf::from_str("/path2").unwrap(),
+        ]);
+    }
+
+    #[test]
+    fn manpath_without_trailing_colon_gets_trailing_colon() {
+        let env_dirs = "/foo:/bar";
+        let manpath = "/baz:/qux";
+        let new_manpath = fix_manpath_var(env_dirs, manpath);
+        assert!(new_manpath.ends_with(':'));
+    }
+
+    #[test]
+    fn manpath_with_trailing_colon_doesnt_get_new_trailing_colon() {
+        let env_dirs = "/foo:/bar";
+        let manpath = "/baz:/qux:";
+        let new_manpath = fix_manpath_var(env_dirs, manpath);
+        assert!(new_manpath.ends_with(':'));
+        assert!(!new_manpath.ends_with("::"));
+    }
+}
```

Comment:
blocking: handle leading colon correctly
Current behavior:
```
> FLOX_ENV_DIRS=/foo:/bar MANPATH=:/baz:/qux FLOX_ENV=/bar ./result/activate.d/flox-env-helper zsh | grep MANPATH
export MANPATH="/foo/share/man:/bar/share/man::/baz:/qux";
```

Test failing with the new behavior:
```
    #[test]
    fn manpath_with_leading_colon_gets_two_colons() {
        let env_dirs = "/foo:/bar";
        let manpath = ":/baz:/qux";
        let new_manpath = fix_manpath_var(env_dirs, manpath);
        assert_eq!("/foo/share/man:/bar/share/man::/baz:/qux", new_manpath);
    }
```

Reply:
Done